src/api/rpc_api/events.rs

```diff
@@ -0,0 +1,372 @@
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+	   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+use crate::{
+	api::{Api, Error, Result},
+	rpc::{HandleSubscription, Request, Subscribe},
+	utils, FromHexString, GetBlock, GetStorage, Phase,
+};
+use ac_node_api::{EventDetails, Events, StaticEvent};
+use ac_primitives::{ExtrinsicParams, FrameSystemConfig, StorageChangeSet};
+use alloc::{string::ToString, vec::Vec};
+use codec::Encode;
+use log::*;
+use serde::de::DeserializeOwned;
+use sp_runtime::traits::{Block as BlockTrait, GetRuntimeBlockType, Hash as HashTrait};
+pub trait FetchEvents<Client, Hash>
+where
+	Client: Request,
+	Hash: DeserializeOwned,
+{
+	/// Fetch all block events from node for the given block hash.
+	fn fetch_events_from_block(&self, block_hash: Hash) -> Result<Events<Hash>>;
+
+	/// Fetch all assosciated events for a given extrinsic hash and block hash.
+	fn fetch_events_for_extrinsic(
+		&self,
+		block_hash: Hash,
+		extrinsic_hash: Hash,
+	) -> Result<Vec<EventDetails>>;
+}
+
+impl<Signer, Client, Params, Runtime> FetchEvents<Client, Runtime::Hash>
+	for Api<Signer, Client, Params, Runtime>
+where
+	Client: Request,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime: FrameSystemConfig + GetRuntimeBlockType,
+	Runtime::RuntimeBlock: BlockTrait + DeserializeOwned,
+	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime::Hash: FromHexString,
+{
+	fn fetch_events_from_block(&self, block_hash: Runtime::Hash) -> Result<Events<Runtime::Hash>> {
+		let key = utils::storage_key("System", "Events");
+		let event_bytes = self
+			.get_opaque_storage_by_key_hash(key, Some(block_hash))?
+			.ok_or(Error::NoBlock)?;
+		let events =
+			Events::<Runtime::Hash>::new(self.metadata().clone(), Default::default(), event_bytes);
+		Ok(events)
+	}
+
+	fn fetch_events_for_extrinsic(
+		&self,
+		block_hash: Runtime::Hash,
+		extrinsic_hash: Runtime::Hash,
+	) -> Result<Vec<EventDetails>> {
+		let extrinsic_index =
+			self.retrieve_extrinsic_index_from_block(block_hash, extrinsic_hash)?;
+		let block_events = self.fetch_events_from_block(block_hash)?;
+		self.filter_extrinsic_events(block_events, extrinsic_index)
+	}
+}
+
+// FIXME: This should rather be implemented directly on the
+// Subscription return value, rather than the api. Or directly
+// subcribe. Should be looked at in #288
+// https://github.com/scs/substrate-api-client/issues/288#issuecomment-1346221653
+pub trait SubscribeEvents<Client, Hash>
```

Comment:
Was not changed at all, simply renamed the file from `subscribe_events` to `events`. This will be updated in a upcoming PR (see issue #413)